pybamm/solvers/casadi_solver.py

```diff
@@ -261,14 +261,15 @@ def _integrate(self, model, t_eval, inputs_dict=None):
                     except pybamm.SolverError as error:
                         pybamm.logger.debug("Failed, halving step size")
                         dt /= 2
+                        count += 1
                         # also reduce maximum step size for future global steps,
                         # but skip them in the beginning
                         # sometimes, for the first integrator smaller timesteps are
                         # needed, but this won't affect the global timesteps. The
                         # global timestep will only be reduced after the first timestep.
                         if first_ts_solved:
                             dt_max = dt
-                        if count >= self.max_step_decrease_count:
+                        if count - 1 >= self.max_step_decrease_count:
```

Comment:
would just doing `count > self.max_step_decrease_count` be simpler to read?

Reply:
Just doing `count > self.max_step_decrease_count` would be simpler to read, I have changed it.
The "parameter-values" notebook runs locally for me, but I noticed that there was a different value assigned for a parameter (b) in the markdown cell than in the code cell of the "Solving a model" section and assigned it to the value of the code cell.